### ISSUE TITLE
Non-blocking push?

### ISSUE THREAD

**Issue by OrangeDog** [rigtorp/SPSCQueue#42]
It it possible to tweak this such that the `push` methods do not block when full, but instead pop from the other end like a classic ring buffer?

If it requires significant changes, perhaps you could publish that separately?

**Comment by kallala** [comment]
I dont think this is possible without severely compromising performance or introducting a race condition
